src/index.js

```diff
@@ -45,54 +45,46 @@ export function mapPropsToState(props: Object, state: Object): Object {
   }, {});
 }
 
-// Default HOC options.
-const defs: Options = {
-  mapDefaultPropsToProps,
-  mapPropsToState
+type Props = {
+  children: Function,
+  component: Object,
```

Comment:
Passing in the whole component so both its state and props can be accessed is neat but it doesn't solve the problem where you want to restrict only certain parts of state to be controlled via props.

What about passing in the component props and state as separate props to react-ctrl which would then allow the consumer to restrict props if they wished to?

Reply:
I'm trying to keep the API as thin as possible. If you want to only control some of the state, can't you just access state from the component itself via `this.state`?

Reply:
See below for some ideas around this :). In retrospect it might make it simpler, even if you have to pass two props.